source/docs/subset_load_balancer.md

```diff
@@ -0,0 +1,273 @@
+
+Given this CDS `envoy::api::v2::Cluster`:
+
+``` json
+{
+  "name": "c1",
+  "lb_policy": "ROUND_ROBIN",
+  "lb_subset_config": {
+    "fallback_policy": "DEFAULT_SUBSET",
+    "default_subset": {
+      "stage": "prod",
+      "version": "1.0",
+      "type": "std"
+    },
+    "subset_selectors": [
+      { "keys": [ "stage", "type" ] },
+      { "keys": [ "stage", "version" ] },
+      { "keys": [ "version" ] },
+      { "keys": [ "xlarge", "version" ] },
+    ]
+  }
+}
+```
+
+The following subsets are created:
+
+`stage=prod, type=std` (e1, e2, e3, e4)
+`stage=prod, type=bigmem` (e5, e6)
+`stage=dev, type=std` (e7)
+`stage=prod, version=1.0` (e1, e2, e5)
+`stage=prod, version=1.1` (e3, e4, e6)
+`stage=dev, version=1.2-pre` (e7)
+`version=1.0` (e1, e2, e5)
+`version=1.1` (e3, e4, e6)
+`version=1.2-pre` (e7)
+`version=1.0, xlarge=true` (e1, e3)
+
+In addition, a default subset is created:
+
+`stage=prod, type=std, version=1.0` (e1, e2)
+
+After loading this configuration, the SLB's `LbSubsetMap` looks like this:
+
+<a name="diagram"></a>
+![LbSubsetMap Diagram](subset_load_balancer_diagram.svg)
+
+Given these `envoy::api::v2::Route` entries:
+
+``` json
+"routes": [
+  {
+    "match": {
+      "prefix": "/",
+      "headers": [
+        {
+          "name": "x-custom-version",
+          "value": "1.2-pre"
+        }
+      ]
+    },
+    "route": {
+      "cluster": "c1",
+      "metadata_match": {
+        "filter_metadata": {
+          "envoy.lb": {
+            "version": "1.2-pre",
+            "stage": "dev"
+          }
+        }
+      }
+    }
+  },
+  {
+    "match": {
+      "prefix": "/",
+      "headers": [
+        {
+          "name": "x-hardware-test",
+          "value": "bigmem"
+        }
+      ]
+    },
+    "route": {
+      "cluster": "c1",
+      "metadata_match": {
+        "filter_metadata": {
+          "envoy.lb": {
+            "type": "bigmem",
+            "stage": "prod"
+          }
+        }
+      }
+    }
+  },
+  {
+    "match": {
+      "prefix": "/"
+    },
+    "route": {
+      "weighted_clusters": {
+        "clusters": [
+          {
+            "name": "c1",
+            "weight": 90,
+            "metadata_match": {
+              "filter_metadata": {
+                "envoy.lb": {
+                  "version": "1.0"
+                }
+              }
+            }
+          },
+          {
+            "name": "c1",
+            "weight": 10,
+            "metadata_match": {
+              "filter_metadata": {
+                "envoy.lb": {
+                  "version": "1.1"
+                }
+              }
+            }
+          }
+        ]
+      },
+      "metadata_match": {
+        "filter_metadata": {
+          "envoy.lb": {
+            "stage": "prod",
+          }
+        }
+      }
+    }
+  }
+]
+```
+
+The following headers may then be used to select subsets:
+
+`x-custom-version: 1.2-pre` causes requests to be routed e7. This is an example of routing requests
```

Comment:
You might want to add that these headers have nothing to do with lb.metadata names you defined above. OR for clarity, you could change the header values from 1.2-pre to something else, as during first read, it looks like you can specify the metadata selectors in http headers [while I would love that, its not the focus of this doc :) ]

Reply:
I'll make it clear they don't have to match.